### PR TITLE
Full archive heartbeat sender + new data pools

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- further integration for the full archive network 
  
## Proposed changes
- update heartbeatV2Components to send the info on the second network

## Testing procedure
- will be tested with the full feature

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: heartbeat/sender/multikeyHeartbeatSender.go [hunk 1]
```diff
@@ -184,7 +185,8 @@ func (sender *multikeyHeartbeatSender) sendMessageForKey(pkBytes []byte) error {
 		return err
 	}
 
-	sender.messenger.BroadcastUsingPrivateKey(sender.topic, buff, pid, p2pSk)
+	sender.mainMessenger.BroadcastUsingPrivateKey(sender.topic, buff, pid, p2pSk)
+	sender.fullArchiveMessenger.BroadcastUsingPrivateKey(sender.topic, buff, pid, p2pSk)
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply heartbeat/sender/multikeyHeartbeatSender.go:189]: added

File: heartbeat/sender/multikeyPeerAuthenticationSender_test.go [hunk 2]
```diff
@@ -144,25 +143,37 @@ func createMockMultikeyPeerAuthenticationSenderArgsSemiIntegrationTests(
...
-		assert.True(t, check.IfNil(senderInstance))
-		assert.Equal(t, heartbeat.ErrNilMessenger, err)
+		assert.Nil(t, senderInstance)
+		assert.True(t, errors.Is(err, heartbeat.ErrNilMessenger))
+	})
+	t.Run("nil full archive messenger should error", func(t *testing.T) {
+		t.Parallel()
+
+		argsBase := createMockBaseArgs()
+		argsBase.fullArchiveMessenger = nil
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply heartbeat/sender/multikeyPeerAuthenticationSender_test.go:170]: the fullArchiveMessenger is part of the baseSender, used in both heartbeat and peerAuthentication.. nil checks are done as part of the baseSender

File: heartbeat/sender/commonPeerAuthenticationSender.go [hunk 3]
```diff
@@ -40,12 +40,12 @@ func (cpas *commonPeerAuthenticationSender) generateMessageBytes(
 	msg.Payload = payloadBytes
 
 	if p2pSkBytes != nil {
-		msg.PayloadSignature, err = cpas.messenger.SignUsingPrivateKey(p2pSkBytes, payloadBytes)
+		msg.PayloadSignature, err = cpas.mainMessenger.SignUsingPrivateKey(p2pSkBytes, payloadBytes)
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply heartbeat/sender/commonPeerAuthenticationSender.go:43]: yes, as this message won't be sent on the full archive network anyway.. mainMessenger = the one from main network

File: factory/heartbeat/heartbeatV2Components.go [hunk 4]
```diff
@@ -247,42 +246,95 @@ func (hcf *heartbeatV2ComponentsFactory) Create() (*heartbeatV2Components, error
...
+		shardSender:                          shardSender,
+		monitor:                              heartbeatsMonitor,
+		statusHandler:                        statusHandler,
+		mainDirectConnectionProcessor:        mainDirectConnectionProcessor,
+		fullArchiveDirectConnectionProcessor: fullArchiveDirectConnectionProcessor,
 	}, nil
 }
 
+func (hcf *heartbeatV2ComponentsFactory) createTopicsIfNeeded() error {
+	err := createTopicIfNeededOnMessenger(hcf.networkComponents.NetworkMessenger())
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply factory/heartbeat/heartbeatV2Components.go:313]: done

File: heartbeat/sender/commonPeerAuthenticationSender.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply heartbeat/sender/commonPeerAuthenticationSender.go:43]: ok

**Review by iulianpascalau — APPROVED** [review]

File: heartbeat/sender/multikeyPeerAuthenticationSender_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply heartbeat/sender/multikeyPeerAuthenticationSender_test.go:170]: ok

**Review by ssd04 — APPROVED** [review]

**Pull request merged**